feat: Rebrand UI 💎 ✨🎊

surajshetty3416 opened the pull request:
## New Design

#### Login Page
<img width="800" alt="Screenshot 2021-01-28 at 5 22 00 PM" src="https://user-images.githubusercontent.com/13928957/106147523-f69a5000-619d-11eb-8034-e46c0493605f.png">

#### Workspace
<img width="800" alt="Screenshot 2021-02-01 at 2 55 45 PM" src="https://user-images.githubusercontent.com/13928957/106439358-9b1dda00-649d-11eb-8102-da5156413046.png">



#### List View
<img width="800" alt="Screenshot 2021-01-28 at 5 24 02 PM" src="https://user-images.githubusercontent.com/13928957/106147481-e97d6100-619d-11eb-8200-13a715df6d71.png">

#### Form View
<img width="800" alt="Screenshot 2021-01-28 at 7 26 10 PM" src="https://user-images.githubusercontent.com/13928957/106148168-ba1b2400-619e-11eb-93bc-cca52ba3a583.png">

<img width="800" alt="Screenshot 2021-01-28 at 7 47 14 PM" src="https://user-images.githubusercontent.com/13928957/106150757-ab823c00-61a1-11eb-905c-ed8f6286bb98.png">

#### Report View
<img width="800" alt="Screenshot 2021-01-28 at 5 24 38 PM" src="https://user-images.githubusercontent.com/13928957/106147475-e71b0700-619d-11eb-81f2-13cef90fe56c.png">

#### Print View
<img width="800" alt="Screenshot 2021-01-28 at 5 23 25 PM" src="https://user-images.githubusercontent.com/13928957/106147809-5133ac00-619e-11eb-85b9-caccd6d02f40.png">

#### Kanban
<img width="800" alt="Screenshot 2021-01-28 at 7 45 07 PM" src="https://user-images.githubusercontent.com/13928957/106150413-57775780-61a1-11eb-8fa1-e72f5f13af58.png">

#### File Uploader
<img width="800" alt="Screenshot 2021-01-28 at 7 24 51 PM" src="https://user-images.githubusercontent.com/13928957/106147970-8213e100-619e-11eb-9fa3-aea69280ecc9.png">

## Dark Mode 🌙
`(cmd + shift + g)`

<img width="1440" alt="Screenshot 2021-02-01 at 2 50 47 PM" src="https://user-images.githubusercontent.com/13928957/106438753-eaafd600-649c-11eb-94ef-37fd8bb0d428.png">


## New Routing 
using [history](https://developer.mozilla.org/en-US/docs/Web/API/History) API to set cleaner URLs

**Before:**
>https://\<site\>.erpnext.com/desk#List/User/List
>https://\<site\>.erpnext.com/desk#Form/User/test@erpnext.com
>https://\<site\>.erpnext.com/desk#List/User/Report

**Now:**
>https://\<site\>.erpnext.com/app/user
>https://\<site\>.erpnext.com/app/user/test@erpnext.com
>https://\<site\>.erpnext.com/app/user/view/report

### Updated Cypress to v6

- For [performance improvements](https://docs.cypress.io/guides/references/changelog.html#5-2-0), [auto-retry feature for flaky tests](https://docs.cypress.io/guides/references/changelog.html#5-0-0) and [to fix recording issue](https://github.com/cypress-io/cypress/issues/9265#issuecomment-735984040)


---

Co-authored-by: Prssanna Desai <prssud@gmail.com>
Co-authored-by: Shivam Mishra <scm.mymail@gmail.com>
Co-authored-by: Rushabh Mehta <rmehta@gmail.com>
Co-authored-by: Faris Ansari <netchamp.faris@gmail.com>

> no-docs

BREAKING CHANGE

surajshetty3416 committed:
fix: Link Icon active color

surajshetty3416 committed:
fix: Heatmap color in dark mode

surajshetty3416 committed:
fix: Set radius for heatmap boxes

surajshetty3416 committed:
fix: Update feeze backdrop color

- changes based on theme

surajshetty3416 committed:
fix: Awesomplete input color

surajshetty3416 committed:
fix: Attach style

surajshetty3416 committed:
fix: Icon color for dark mode

surajshetty3416 committed:
fix: Quill icon stroke width

surajshetty3416 committed:
fix: Support routes with desk in it

surajshetty3416 committed:
fix: Expire session only if non-guest gets logged out
- fix formatting

surajshetty3416 committed:
fix: Quill container style

surajshetty3416 committed:
fix: Color picker UX [minor]

surajshetty3416 committed:
feat: Show email status in message box

- Show link of docuement in message box

surajshetty3416 committed:
fix: Use indicator pill instead of plain indicator

surajshetty3416 committed:
fix: Set title attribute for page title

scmmishra committed:
fix: phone control top position

scmmishra committed:
fix: awesomplete options hover bg in dark mode

scmmishra committed:
feat: don't show avatar for comments

scmmishra committed:
fix: typo comment

scmmishra committed:
feat: spacing in timeline comment box

rmehta committed:
fix(minor): validate links in customize form

rmehta committed:
fix(minor): fix link in User doctype

scmmishra committed:
fix: remove console statement

rmehta committed:
fix(tests): routing, permission and other fixes

rmehta committed:
fix(minor): tests + routing

prssanna committed:
refactor: common css for website and desk

scmmishra committed:
fix: navbar and breadcrumb bg

scmmishra committed:
feat: allow adding a class to HTML body

rmehta committed:
fix(tests): add a timeout since, faster execution?

prssanna committed:
fix: disable hidden for enabled checkbox in User

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
refactor: filter popover behaviour

apply filter on input focusout

rmehta committed:
fix(conflict)

prssanna committed:
fix: group by popover hashchange event

prssanna committed:
fix: bind dropdown event to added sidebar filters

rmehta committed:
Merge pull request #12119 from rmehta/validate-link-for-customize

fix(minor): Link Validation in Customize Form

prssanna committed:
fix: support for filters in set_route

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

scmmishra committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: cypress tests

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

rmehta committed:
fix(routing): removed /space from routing

rmehta committed:
fix(minor): homepage and tests

rmehta committed:
fix(minor): javascript lint

rmehta committed:
fix(minor): homepage for frappe and tests

rmehta committed:
Merge pull request #12143 from rmehta/workspace-routing

fix(routing): removed /space from routing

rmehta committed:
feat(minor): Build Workspace, renamed frappe.views.Views => frappe.views.ListViewSelect, fixed recurring dependency issue on grids

prssanna committed:
fix: overflow in address box

prssanna committed:
fix: progress bar background color

gavindsouza committed:
fix: Strip HTML only if string is passed, else evaluate like before (#12157)

rmehta committed:
Merge pull request #12145 from rmehta/build-workspace

feat(minor): Build Workspace, renamed frappe.views.Views => frappe.views.ListViewSelect, fixed recurring dependency issue on grids

hasnain2808 committed:
fix: strip_html breaks when it gets undefined

surajshetty3416 committed:
fix: Update grays for better contrast

surajshetty3416 committed:
fix: Update disabled-text-color

surajshetty3416 committed:
fix: Remove unnecessary z-index from phone icon

rmehta committed:
fix(minor): merge develop

surajshetty3416 committed:
fix: Remove unnecessary margin left

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: page form layout

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Disable comment box till the comment is created on server-side

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Comment update bug

Do not wait for socketio to update owner's timeline... update immediately.

surajshetty3416 committed:
fix: Sync comment updates realtime

surajshetty3416 committed:
fix: Show owner name for versions of child tables

prssanna committed:
fix: add Tree to view modes

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: disabled attribute for disabled checkboxes

prssanna committed:
fix: disabled checkbox design

prssanna committed:
fix: re route issue for new docs

surajshetty3416 committed:
fix: Comment unhandled code

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Sync comment update realtime

rmehta committed:
fix(minor): redirect to login if original route is broken

rmehta committed:
Merge pull request #12192 from rmehta/fix-guest-routing

fix(minor): redirect to login if original route is broken

rmehta committed:
BREAKING CHANGE: rename child tables of Workspace

rmehta committed:
fix(minor): revert force reloading of workspace

prssanna committed:
fix: remove disabled attribute from disabled checkbox

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: disable print margin in ace editor

surajshetty3416 committed:
fix: Keep email.css inside frappe assets

rmehta committed:
fix(minor): typo

rmehta committed:
Merge pull request #12195 from rmehta/rename-workspace-child-tables

BREAKING CHANGE: rename child tables of Workspace

surajshetty3416 committed:
refactor: Move email.less to email.scss

- Updated email container layout & btn style

surajshetty3416 committed:
fix: Update preview_email function to return only HTML

surajshetty3416 committed:
feat: Update standard email template to match new style

surajshetty3416 committed:
feat: Set frappe.bold as safe global

surajshetty3416 committed:
refactor: Move verification code email msg to template

surajshetty3416 committed:
fix: Update workflow action email template style

surajshetty3416 committed:
fix: Add header to workflow action

prssanna committed:
random

surajshetty3416 committed:
fix: Get brand logo from app

surajshetty3416 committed:
fix: Update new-user email template style

surajshetty3416 committed:
fix: Update energy points summary style

surajshetty3416 committed:
fix: Report routing issue

rmehta committed:
fix(minor): setup-wizard page_name shows as undefined

rmehta committed:
fix(minor): keep form dashboard sections expanded by default

rmehta committed:
fix(minor): ui fixes to grid scrolling, etc

rmehta committed:
fix(minor): background in base.html

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Update email template styles

surajshetty3416 committed:
fix: Update email container padding

surajshetty3416 committed:
fix: Add alt and title to brand logo

prssanna committed:
fix: overflow hidden for text editor in print view

surajshetty3416 committed:
fix: Email body width and padding

rmehta committed:
fix(minor): clear all_apps cache

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

rmehta committed:
fix(minor): setting default doctype in customize form

prssanna committed:
fix: remove overflow-x for editor in print

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Trigger timeline refresh for backward compatibility

surajshetty3416 committed:
fix: Add doctype and docname to timeline item

surajshetty3416 committed:
feat: Add branch icon

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Email template styles

surajshetty3416 committed:
feat: Add read-status & unread-status icons

surajshetty3416 committed:
fix: Text editor tooltip hidding issue

- Fix tooltip style

surajshetty3416 committed:
fix: Listview setting header refresh issue

surajshetty3416 committed:
fix: Prioritize reply communication_doc sender

- Prioritize reply communication_doc sender email over form email

surajshetty3416 committed:
Merge branch 'develop' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Post merge fixes

surajshetty3416 committed:
fix: Post merge error

surajshetty3416 committed:
fix: Timeline action style

surajshetty3416 committed:
fix: Email new notification style

surajshetty3416 committed:
fix: Email footer style

prssanna committed:
fix: website fixes

-move modal and icon styles to common

primary navbar theme

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
refactor: Fix address naming

prssanna committed:
fix: kanban switcher in page custom actions

prssanna committed:
fix: dropdown divider style

surajshetty3416 committed:
feat: Add avatar macro

surajshetty3416 committed:
feat: Add with_container argument and update email style

surajshetty3416 committed:
feat: Add avatars for top performers

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Post merge issue

prssanna committed:
fix: escape kanban name in data attribute

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Get brand logo from email account

surajshetty3416 committed:
fix: Show brand logo only if header or with_container is set

surajshetty3416 committed:
chore: Update frappe-charts

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Show brand logo if header is set

surajshetty3416 committed:
fix: Relaod docperm doctype to sync recent changes

surajshetty3416 committed:
fix: Email body spacing issue

surajshetty3416 committed:
fix: Remove fallback title for brand logo

surajshetty3416 committed:
fix: Responsive email style

surajshetty3416 committed:
fix: Blockquote style

surajshetty3416 committed:
fix: Use get_abbr method for abbr

surajshetty3416 committed:
refactor: Introduce variables for error state

surajshetty3416 committed:
Merge branch 'develop' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Handle edge case in get_abbr method

surajshetty3416 committed:
fix(Patch): Rename Desk Link only if it exists

surajshetty3416 committed:
fix: Typo

surajshetty3416 committed:
fix: Incorrect fix

surajshetty3416 committed:
fix: Keep indicator as first element

surajshetty3416 committed:
fix: Select control style

prssanna committed:
fix: reorganize styles

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: remove duplicate style

surajshetty3416 committed:
refactor: Move default email.css to the root css folder

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Pass with_container parameter value

surajshetty3416 committed:
feat: Add icon to timeline action button

surajshetty3416 committed:
fix: Stat indicator style in shortcut

surajshetty3416 committed:
fix: Show content in right timeline

- Previously, it used to show content of other form's
timeline in another form's timeline

surajshetty3416 committed:
fix(form): Check for action type instead of key

surajshetty3416 committed:
fix: Temporarily comment buggy code to avoid confusion

hasnain2808 committed:
fix: Validate reqd report filters (#12230)

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

netchampfaris committed:
fix: file browser in uploader

- use new icons
- server side search
- pagination for files in folder

rmehta committed:
fix(minor): weird grid scrolling behaviour

surajshetty3416 committed:
fix: Email server-side tests

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: ui tests

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: common indicator style for website and desk

prssanna committed:
fix: revert login password changes

prssanna committed:
fix: make add filter a button

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Add map defaults

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Recorder routing

surajshetty3416 committed:
Merge branch 'develop' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
test: Enable tests for recorder

surajshetty3416 committed:
style: Fix sider issues

surajshetty3416 committed:
test: Import unquote

surajshetty3416 committed:
test: Fix newsletter test

prssanna committed:
fix: list view ui test

prssanna committed:
fix: ui tests

prssanna committed:
style: spacing issues

prssanna committed:
style: brace formatting issues

surajshetty3416 committed:
fix: Remove tnum variation of zero
- to avoid readability issue

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
style: indent issues

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
style: Fix formatting

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
chore: Add .stylelintrc file to add rules

ghost commented:
✅ Errors or warnings were resolved via the commit 94db4b6! Please visit [Sider](https://sider.review/gh/repos/1864194/pulls/12277) for details.

---

You can [turn off this notification](https://sider.review/gh/repos/1864194/settings#notifications) if you don't want to receive it from now on.

surajshetty3416 committed:
chore: Fix syntax in .stylelintrc

prssanna committed:
style: indent issues

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: disabled cursor for disabld fields

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
style: Fix formatting issues

surajshetty3416 committed:
style: Fix formatting issue

prssanna committed:
fix: set_route

prssanna committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
style: fix formatting

prssanna committed:
style: fix formatting

prssanna committed:
style: fix formatting

prssanna committed:
fix: duplicate width height style

prssanna committed:
fix: list view ui test

prssanna committed:
fix: ignore no descending specifity rule

prssanna committed:
fix: typo

prssanna committed:
fix: typo

prssanna committed:
fix: letterhead selector issue in print dialog

prssanna committed:
fix: formatting

surajshetty3416 committed:
style: Fix formatting issues

surajshetty3416 committed:
style: Fix formatting issue

surajshetty3416 committed:
style: Fix duplicate selector issue

surajshetty3416 committed:
style: Fix duplicate selector issue

surajshetty3416 committed:
chore: Fix translation syntax check

surajshetty3416 committed:
style: Fix translation syntax

surajshetty3416 committed:
chore: Show correct line number

surajshetty3416 committed:
chore: Fix translation syntax check

surajshetty3416 committed:
chore: Update cypres project & record ID

rmehta committed:
fix(minor): fix routing and titles + added frappe.utils.markdown alias to frappe.utils.md_to_html

surajshetty3416 committed:
revert: "chore: Update cypres project & record ID"

This reverts commit 4148fbfad50f754fc198aaa0d9d43e342d25f496.

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
test: Fix recorder tests

surajshetty3416 committed:
style: Remove unused variable

surajshetty3416 committed:
fix: Close <li> tag

- Fix formatting

surajshetty3416 committed:
chore: Update Cypress 3 to 6 & fix test cases

surajshetty3416 committed:
test: Make tests Cypress v6 compatible

surajshetty3416 committed:
style: Fix indentation

surajshetty3416 committed:
test: Fix recorder and datetime tests

surajshetty3416 committed:
chore: Enable cypress retries

surajshetty3416 committed:
chore: Reinstall producer site only for server-side tests

surajshetty3416 committed:
chore: Fix syntax error

surajshetty3416 committed:
chore: Skip porducer site installation in ui test

surajshetty3416 committed:
test: Fix failing tests

surajshetty3416 committed:
Merge branch 'develop' of https://github.com/frappe/frappe into rebrand-ui

prssanna committed:
fix: indicator ellipsis on desk

surajshetty3416 committed:
fix: Rename Desk Shortcut to Workspace Shortcut

surajshetty3416 committed:
Merge branch 'rebrand-ui' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Remove margin bottom from last form-control

surajshetty3416 committed:
fix: Typo

surajshetty3416 committed:
Merge branch 'develop' of https://github.com/frappe/frappe into rebrand-ui

surajshetty3416 committed:
fix: Frappe control style

surajshetty3416 committed:
fix: Update User modified

to sync changes made in following PR
https://github.com/frappe/frappe/pull/12148

prssanna committed:
Merge branch 'develop' of https://github.com/frappe/frappe into rebrand-ui

iMoshi commented:
Quick heads up, getting `frappe.core.doctype.doctype.doctype.InvalidFieldNameError: Row #1: Could not find field <b>web_template</b> in <b>Web Page</b> DocType` :)

centaur78 commented:
I am getting this error during bench update now

```
Executing execute:frappe.rename_doc("Workspace", "Loan", "Loan Management", ignore_if_exists=True, force=True) in site1.local (_1bd3e0294da19198)
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 99, in <module>
    main()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 18, in main
    click.Group(commands=commands)(prog_name='bench')
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/commands/__init__.py", line 26, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/commands/site.py", line 296, in migrate
    skip_search_index=skip_search_index
  File "/home/frappe/frappe-bench/apps/frappe/frappe/migrate.py", line 67, in migrate
    frappe.modules.patch_handler.run_all(skip_failing)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 41, in run_all
    run_patch(patch)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 30, in run_patch
    if not run_single(patchmodule = patch):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 71, in run_single
    return execute_patch(patchmodule, method, methodargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 89, in execute_patch
    exec(patchmodule.split("execute:")[1],globals())
  File "<string>", line 1, in <module>
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 891, in rename_doc
    return rename_doc(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/rename_doc.py", line 63, in rename_doc
    rename_parent_and_child(doctype, old, new, meta)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/rename_doc.py", line 186, in rename_parent_and_child
    update_child_docs(old, new, meta)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/rename_doc.py", line 240, in update_child_docs
    % (df.options, '%s', '%s'), (new, old))
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/database.py", line 147, in sql
    self._cursor.execute(query, values)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/pymysql/cursors.py", line 170, in execute
    result = self._query(query)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/pymysql/cursors.py", line 328, in _query
    conn.query(q)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/pymysql/connections.py", line 517, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/pymysql/connections.py", line 732, in _read_query_result
    result.read()
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/pymysql/connections.py", line 1075, in read
    first_packet = self.connection._read_packet()
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/pymysql/connections.py", line 684, in _read_packet
    packet.check_error()
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/pymysql/protocol.py", line 220, in check_error
    err.raise_mysql_exception(self._data)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/pymysql/err.py", line 109, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.ProgrammingError: (1146, "Table '_1bd3e0294da19198.tabWorkspace Link' doesn't exist")
```

centaur78 commented:
Hi @surajshetty3416 

The UI seems to break when you are trying to create the purchase order from sales order. I have opened the issue here [https://github.com/frappe/erpnext/issues/24533](url). Really appreciate your support

![setting_dependancy](https://user-images.githubusercontent.com/1904802/106844582-59ba4400-66c2-11eb-98a3-f205eb972bc9.jpg)
